File: pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    text = get_string("Text: ")

    l_c = letters_counter(text)
    w_c = words_counter(text)
    s_c = sentences_counter(text)

    # Calculation of the Coleman-Liau Index
    L = l_c * 100 / w_c
    S = s_c * 100 / w_c
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # print the results
    if index < 1:
        print("Before Grade 1")

    elif index > 12:
        print("Grade 16+")

    else:
        print(f"Grade {index}")


# count the number of letters


def letters_counter(text):
    l_c = 0
    for c in text:
        if c.lower() >= "a" and c.lower() <= "z":
            l_c += 1

    return l_c

# count the number of words


def words_counter(text):
    w_c = 0
    for c in text:
        if c == ' ':
            w_c += 1

    return w_c + 1

# count the number of sentences


def sentences_counter(text):
    s_c = 0
    for c in text:
        if c == '!' or c == '.' or c == '?':
            s_c += 1

    return s_c


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    //Get a text
    string text = get_string("Text: ");

    float nl = count_letters(text);
    float nw = count_words(text);
    float ns = count_sentences(text);
    //Calculation of the Coleman-Liau Index
    float L = 100 * nl / nw;
    float S = 100 * ns / nw;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    //print the results
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

//count the number of letters
int count_letters(string text)
{
    int k = 0;
    for (int i = 0, n = strlen(text); n > i ; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            k++;
        }
    }
    return k;
}
//count the number of words
int count_words(string text)
{
    int w = 0;
    for (int i = 0, n = strlen(text); n > i ; i++)
    {
        if (text[i] == ' ')
        {
            w++;
        }
    }
    return w + 1;
}
//count the number of sentences
int count_sentences(string text)
{
    int s = 0;
    for (int i = 0, n = strlen(text); n > i ; i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            s++;
        }
    }
    return s;
}